pg_shard.c

```diff
@@ -258,21 +284,49 @@ PgShardPlanner(Query *query, int cursorOptions, ParamListInfo boundParams)
 		selectFromMultipleShards = SelectFromMultipleShards(query, queryShardList);
 		if (selectFromMultipleShards)
 		{
-			Oid distributedTableId = InvalidOid;
 			Query *localQuery = NULL;
 			List *queryRestrictList = QueryRestrictList(distributedQuery);
 			List *remoteRestrictList = NIL;
 			List *localRestrictList = NIL;
+			bool localQueryHasAggregates = false;
+			bool remoteQueryHasAggregates = false;
+			bool aggregatesPushedDown = false;
 
 			/* partition restrictions into remote and local lists */
 			ClassifyRestrictions(queryRestrictList, &remoteRestrictList,
 								 &localRestrictList);
 
+			/* decide remote and local hasAggs values */
+			ClassifyHasAggregates(query, &remoteQueryHasAggregates,
+								  &localQueryHasAggregates);
+
 			/* build local and distributed query */
-			distributedQuery = RowAndColumnFilterQuery(distributedQuery,
-													   remoteRestrictList,
-													   localRestrictList);
-			localQuery = BuildLocalQuery(query, localRestrictList);
+			distributedQuery = BuildDistributedQuery(distributedQuery,
+													 remoteRestrictList,
+													 localRestrictList,
+													 remoteQueryHasAggregates);
+			localQuery = BuildLocalQuery(query, localRestrictList,
+										 localQueryHasAggregates,
+										 remoteQueryHasAggregates);
+
+			/* determine whether we pushed down aggregates or not */
```

Comment:
Merge this comment into the block comment below somehow and move the assignment to be right above the `if` statement.